Introduce FHMAX_ENKF_GFS to extending ensemble forecast capabilities

XianwuXue-NOAA opened the pull request:
<!-- PLEASE READ -->
<!--
Before opening a PR, please note these guidelines:

- Each PR should only address ONE topic and have an associated issue
- No hardcoded or paths to personal directories should be present
- No temporary or backup files should be committed
- Any code that was disabled by being commented out should be removed
-->

**Description**
Introduce FHMAX_ENKF_GFS to extending ensemble forecast capabilities for gfsefmn jobs

<!-- Please include relevant motivation and context. -->
<!-- Please include a summary of the change and which issue is fixed. -->
<!-- List any dependencies that are required for this change. -->

<!-- Please provide reference to the issue this pull request is addressing. -->
<!-- For e.g. Fixes #IssueNumber -->

**Type of change**

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**
Tested it on WCOSS2

<!-- Please describe the tests that you ran to verify your changes and on the platforms these tests were conducted. -->
<!-- Provide instructions so we can reproduce. -->
<!-- Please also list any relevant details for your test configuration -->

<!-- Use the following as a guide to list your tests and delete options that are not relevant. Expand as necessary. -->
<!--
- [ ] Clone and Build tests on WCOSS Dell P3
- [ ] Cycled test on Orion
- [ ] Forecast-only test on Hera
-->
  
**Checklist**

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas

XianwuXue-NOAA committed:
Modify to expand FHMAX_ENKF for gfsefmn jobs

 On branch feature/expand_FHMAX_ENKF
	modified:   parm/config/config.base.emc.dyn

Refs: #1116

XianwuXue-NOAA committed:
Introduce FHMAX_ENKF_GFS to expand the fcst capacity

 On branch feature/expand_FHMAX_ENKF
 Changes to be committed:
	modified:   parm/config/config.base.emc.dyn
	modified:   scripts/exgdas_enkf_fcst.sh
	modified:   workflow/rocoto/workflow_tasks.py

Refs: #1116

XianwuXue-NOAA committed:
Merge branch 'develop' into feature/expand_FHMAX_ENKF

aerorahul commented on scripts/exgdas_enkf_fcst.sh:
```diff
@@ -112,6 +112,12 @@ export FHOUT=${FHOUT_ENKF:-3}
 # model_configure
 export DELTIM=${DELTIM_ENKF:-${DELTIM:-225}}
 export FHMAX=${FHMAX_ENKF:-9}
+if [[ $EUPD_CYC == "BOTH" ]]; then
+    if [[ $CDUMP == "gfs" ]]; then
+        export FHMAXF=$FHMAX_ENKF_GFS
+    fi
+fi
```
```suggestion
  if [[ ${CDUMP} == "gfs" ]]; then
      export FHMAX=${FHMAX_ENKF_GFS:-${FHMAX_ENKF:-9}}
  fi
```

aerorahul commented on workflow/rocoto/workflow_tasks.py:
```diff
@@ -1192,8 +1192,13 @@ def echgres(self):
+
             fhmin = epos['FHMIN_ENKF']
             fhmax = epos['FHMAX_ENKF']
+            if eupd_cyc == "both":
+                if self.cdump == "gfs":
+                    fhmax = epos['FHMAX_ENKF_GFS']
```
```suggestion
              if self.cdump == "gfs":
                  fhmax = epos['FHMAX_ENKF_GFS']
```

aerorahul reviewed (changes requested):
looks good.
some suggestions to remove superfluous if-blocks.

XianwuXue-NOAA committed:
Remove EUPD_CYC if conditions

 On branch feature/expand_FHMAX_ENKF
	modified:   scripts/exgdas_enkf_fcst.sh
	modified:   workflow/rocoto/workflow_tasks.py

Refs: #1116

XianwuXue-NOAA committed:
Delete unused line

 On branch feature/expand_FHMAX_ENKF
	modified:   workflow/rocoto/workflow_tasks.py

Refs: #1116

XianwuXue-NOAA commented:
@aerorahul Thanks. Delete the if conditions.

aerorahul commented on scripts/exgdas_enkf_fcst.sh:
```diff
@@ -112,6 +112,10 @@ export FHOUT=${FHOUT_ENKF:-3}
 export DELTIM=${DELTIM_ENKF:-${DELTIM:-225}}
 export FHMAX=${FHMAX_ENKF:-9}
+if [[ $CDUMP == "gfs" ]]; then
+   export FHMAXF=$FHMAX_ENKF_GFS
```
There is a typo here on the LHS.
Also, suggest this:
```suggestion
   export FHMAX=${FHMAX_ENKF_GFS:-${FHMAX_ENKF:-${FHMAX}}}
```

XianwuXue-NOAA committed:
Fix a bug and improve it

 On branch feature/expand_FHMAX_ENKF
	modified:   scripts/exgdas_enkf_fcst.sh

Refs: #1116

aerorahul reviewed (approved):
looks good.
